src/Facets/CalldataVerificationFacet.sol

```diff
@@ -100,6 +100,35 @@ contract CalldataVerificationFacet {
         );
     }
 
+    // @notice Extracts the non-EVM address from the calldata
+    // @param data The calldata to extract the non-EVM address from
+    // @return nonEVMAddress The non-EVM address extracted from the calldata
+    function extractNonEVMAddress(
+        bytes calldata data
+    ) external pure returns (bytes32 nonEVMAddress) {
+        bytes memory callData = data;
+        ILiFi.BridgeData memory bridgeData = _extractBridgeData(data);
+
+        if (
+            bytes4(data[:4]) == StandardizedCallFacet.standardizedCall.selector
+        ) {
+            // standardizedCall
+            callData = abi.decode(data[4:], (bytes));
+        }
+
+        if (bridgeData.hasSourceSwaps) {
+            assembly {
+                let offset := mload(add(callData, 0x64)) // Get the offset of the bridge specific data
+                nonEVMAddress := mload(add(callData, add(offset, 0x24))) // Get the non-EVM address
```

Comment:
I think some explanation would be helpful to understand how this works.
Does this imply/require that the non-EVM address parameter in the facetData is always at the first(?) position? Or how do we identify it? 
Such rationale should be added here as a comment imho